### PR TITLE
Refactor readwrite-splitting api

### LINKED ISSUE #18587: Consider refactor `readwrite-splitting` Configuration.
Hi community,

Now,  the `readwrite-splitting` configuration are not user-friendly. The master-slave configuration should be intuitive to the user. Instead of the master-slave configuration to props. So we should be refactor it.

`Static Readwrite-splitting before refactor : `

```
rules:
- !READWRITE_SPLITTING
  dataSources:
    readwrite_ds:
      type: Static
      props:
        write-data-source-name: write_ds
        read-data-source-names: read_ds_0,read_ds_1
      loadBalancerName: random
  loadBalancers:
    random:
      type: RANDOM
```

`Dynamic Readwrite-splitting before refactor : `

```
rules:
- !READWRITE_SPLITTING
  dataSources:
    replication_ds:
      type: Dynamic
      props:
        auto-aware-data-source-name: readwrite_ds
        write-data-source-query-enabled: false
- !DB_DISCOVERY
  dataSources:
    readwrite_ds:
      dataSourceNames:
        - ds_0
        - ds_1
        - ds_2
      discoveryHeartbeatName: mgr-heartbeat
      discoveryTypeName: mgr
  discoveryHeartbeats:
    mgr-heartbeat:
      props:
        keep-alive-cron: '0/5 * * * * ?'
  discoveryTypes:
    mgr:
      type: MGR
      props:
        group-name: 558edd3c-02ec-11ea-9bb3-080027e39bd2
```

`Static Readwrite-splitting after refactor : `

```
 rules:
- !READWRITE_SPLITTING
  dataSources:
    readwrite_ds:
      dataSourceStrategy:
        static:
          writeDataSourceName: write_ds
          readDataSourceNames:
            - read_ds_0
            - read_ds_1
```

`Dynamic Readwrite-splitting before refactor : `

```
rules:
- !READWRITE_SPLITTING
  dataSources:
    readwrite_ds:
      dataSourceStrategy:
        dynamic:
          autoAwareDataSourceName: readwrite_ds
          writeDataSourceQueryEnabled: false
- !DB_DISCOVERY
  dataSources:
    readwrite_ds:
      dataSourceNames:
        - ds_0
        - ds_1
        - ds_2
      discoveryHeartbeatName: mgr-heartbeat
      discoveryTypeName: mgr
  discoveryHeartbeats:
    mgr-heartbeat:
      props:
        keep-alive-cron: '0/5 * * * * ?'
  discoveryTypes:
    mgr:
      type: MGR
      props:
        group-name: 558edd3c-02ec-11ea-9bb3-080027e39bd2
```

### PR DESCRIPTION
**Pull request by zjcnb**
For #18587

### CODE REVIEW

**Comment by LeeGuoPing** [comment]
please fix CI first ^_^

**Review by tristaZero — CHANGES_REQUESTED** [review]

File: examples/shardingsphere-example-generator/src/main/resources/template/proxy/feature/readwrite-splitting.ftl [hunk 1]
```diff
@@ -17,10 +17,12 @@
...
     dataSources:
       readwrite_ds:
-        type: Static
-        props:
-          write-data-source-name: write_ds
-          read-data-source-names: read_ds_0, read_ds_1
+        dataSourceStrategy:
+          staticStrategy:
+            writeDataSourceName: write_ds
+            readDataSourceNames:
```
[the remark this answers is not in the document]
Comment (zjcnb) [reply examples/shardingsphere-example-generator/src/main/resources/template/proxy/feature/readwrite-splitting.ftl]: Done.

File: shardingsphere-features/shardingsphere-readwrite-splitting/shardingsphere-readwrite-splitting-core/src/main/java/org/apache/shardingsphere/readwritesplitting/yaml/config/strategy/YamlStaticReadwriteSplittingStrategyConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import org.apache.shardingsphere.infra.yaml.config.pojo.YamlConfiguration;
+
+import java.util.List;
+
+/**
+ * Static Readwrite-splitting strategy configuration for YAML.
+ */
+@Getter
+@Setter
+public class YamlStaticReadwriteSplittingStrategyConfiguration implements YamlConfiguration {
```
Comment (tristaZero) [inline shardingsphere-features/shardingsphere-readwrite-splitting/shardingsphere-readwrite-splitting-core/src/main/java/org/apache/shardingsphere/readwritesplitting/yaml/config/strategy/YamlStaticReadwriteSplittingStrategyConfiguration.java:31]: final class?

File: shardingsphere-features/shardingsphere-readwrite-splitting/shardingsphere-readwrite-splitting-spring/shardingsphere-readwrite-splitting-spring-namespace/src/main/java/org/apache/shardingsphere/readwritesplitting/spring/namespace/tag/strategy/DynamicReadwriteSplittingBeanDefinitionTag.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ * Readwrite-splitting Dynamic bean definition tag.
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class DynamicReadwriteSplittingBeanDefinitionTag {
+    
+    public static final String DYNAMIC_STRATEGY_ROOT_TAG = "dynamic-strategy";
+    
+    public static final String AUTO_AWARE_DATA_SOURCE_NAME = "auto-aware-data-source-name";
+    
+    public static final String WRITE_DATA_SOURCE_QUERY_ENABLED = "write-data-source-query-enabled";
```
Comment (tristaZero) [inline shardingsphere-features/shardingsphere-readwrite-splitting/shardingsphere-readwrite-splitting-spring/shardingsphere-readwrite-splitting-spring-namespace/src/main/java/org/apache/shardingsphere/readwritesplitting/spring/namespace/tag/strategy/DynamicReadwriteSplittingBeanDefinitionTag.java:33]: a props?

**Review by tristaZero — CHANGES_REQUESTED** [review]

**Review by tristaZero — APPROVED** [review]

**Pull request merged**